src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1465,6 +1465,12 @@
       (format "ALTER TABLE reports ADD COLUMN logs %s DEFAULT NULL" jsonb-type)
       (format "ALTER TABLE reports ADD COLUMN resources %s DEFAULT NULL" jsonb-type))))
 
+(defn factset-hash-field-not-nullable
+  []
+  (jdbc/do-commands
+   "UPDATE factsets SET hash='' WHERE hash is NULL"
```

Comment:
So I'm trying to figure out the idea here, behind why the ticket was raised.

I can't help but think that setting this to anything other than a real hash will cause problems with HA for older data that doesn't have one already at migration time. I mean, if you send this content over via HA, won't it generate a different hash for comparison? /cc @mullr